File: backend/entity/CartItem.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type CartItem struct {
	gorm.Model
	AddedDate time.Time `json:"AddedDate"`

	SheetID uint  `json:"SheetID"`
	Sheet   Sheet `gorm:"foreignKey:SheetID"`

	CartID uint `json:"CartID"`
	Cart   Cart `gorm:"foreignKey:CartID"`

}

File: frontend/src/Interface/CartItem.ts
import { CartInterface } from "./Cart";
import { SheetInterface } from "./Sheet";

export interface CartItemInterface {
    ID?: number | undefined;  
    Sheet?: SheetInterface;   
    sheetId?: number;         
    cartId?: number;          
    cart?: CartInterface;     
    addedDate?: string;      
    createdAt?: string;       
    updatedAt?: string;       
}
